package/velox/spark32/pom.xml

```diff
@@ -40,17 +40,22 @@
   <build>
     <plugins>
       <plugin>
-        <artifactId>maven-assembly-plugin</artifactId>
-        <configuration>
-          <finalName>${jar.assembly.name.prefix}-spark${spark32bundle.version}_${scala.binary.version}-${project.version}</finalName>
-        </configuration>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
         <executions>
           <execution>
-            <id>make-assembly</id>
             <phase>package</phase>
             <goals>
-              <goal>single</goal>
+              <goal>shade</goal>
             </goals>
+            <configuration>
+              <relocations>
+                <relocation>
+                  <pattern>com.google.protobuf</pattern>
+                  <shadedPattern>shaded.com.google.protobuf</shadedPattern>
```

Comment:
I suggest use a special `shadedPattern` prefix, such as `io-glutenproject`, `shaded` is too generic and easy to conflict with other projects

Reply:
Fair enough to me. If the PoC can be proved feasible then we'll use a special pattern than `shaded` in the final patch.